### PR TITLE
Data Processing PR

### PR DESCRIPTION
**Pull request by Javed-Shahbaz**
## In this milestone:
- Set up associations between models.
- add data in tables
- Create at least 4 posts written by one of the users
- Create at least 6 posts comments for one of the posts
- Use other [CRUD methods](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/active_record_crud.md) to find, update and delete entities

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,2 +1,5 @@
 class User < ApplicationRecord
+  has_many :posts, foreign_key: :author_id
+  has_many :comments, foreign_key: :author_id
+  has_many :likes, foreign_key: :author_id
 end
```
Comment (Whoistolu) [inline app/models/user.rb]: - [ ] [REQUIRED] Kindly ensure that the User model has a method that returns 3 most recent posts as stated in the project requirement.

File: app/models/post.rb [hunk 2]
```diff
@@ -1,3 +1,5 @@
 class Post < ApplicationRecord
-  belongs_to :author
+  belongs_to :author, class_name: 'User', foreign_key: :author_id
+  has_many :comments
+  has_many :likes
 end
```
Comment (Whoistolu) [inline app/models/post.rb]: - [ ] [REQUIRED] Kindly ensure that the Post model has a method that updates the posts counter for a user as stated in the project requirement.
- [ ] [REQUIRED] Also, let's make sure that the Post model has a method that returns the 5 most recent comments for a given post as stated in the project requirement.

File: app/models/comment.rb [hunk 3]
```diff
@@ -1,4 +1,4 @@
 class Comment < ApplicationRecord
-  belongs_to :author
+  belongs_to :author, class_name: 'User', foreign_key: :author_id
   belongs_to :post
 end
```
Comment (Whoistolu) [inline app/models/comment.rb]: -  [ ] [REQUIRED] Kindly ensure that the Comment model has a method that updates the comments counter for a post as stated in the project requirement.

File: app/models/like.rb [hunk 4]
```diff
@@ -1,4 +1,4 @@
 class Like < ApplicationRecord
-  belongs_to :author
+  belongs_to :author, class_name: 'User', foreign_key: :author_id
   belongs_to :post
 end
```
Comment (Whoistolu) [inline app/models/like.rb]: - [ ] [REQUIRED] Let's also make sure that the Like model has a method that updates the likes counter for a post as stated in the project requirement.

**Review by Whoistolu — CHANGES_REQUESTED** [review]
Hi @Javed-Shahbaz 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

### Highlight
🔆 Has `User`, `Post`, `Comment` and `Like` models inside `app/models/` directory
🔆 The associations between models are set up correctly according to the given ERD diagram

## Required Changes ♻️

- [ ]  _Check the inline comments under the review for sections to improve._

### Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

### N/A

Cheers, and Happy coding!

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Reem-lab — APPROVED** [review]
# Hi @Javed-Shahbaz,  👋🏻👋🏻

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights

- [ ]  No Linters errors 🔥
- [ ]  Great job working on the The associations between models💯
- [ ]  Followed GitFlow ⭐🚀
- [ ]  Descrtipive PR⭐



## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**